src/google/protobuf/compiler/objectivec/objectivec_helpers.cc

```diff
@@ -839,15 +840,30 @@ string BuildCommentsString(const SourceLocation& location) {
   while (!lines.empty() && lines.back().empty()) {
     lines.pop_back();
   }
-  string prefix("///");
-  string suffix("\n");
+
+  string prefix;
+  string suffix;
   string final_comments;
+  string epilogue;
+
+  if (prefer_single_line && lines.size() == 1) {
+    prefix = "/** ";
+    suffix = " */\n";
+  } else {
+    prefix = " * ";
+    suffix = "\n";
+    final_comments += "/**\n";
+    epilogue = " **/\n";
+  }
+
   for (int i = 0; i < lines.size(); i++) {
     // HeaderDoc uses '\' and '@' for markers; escape them.
-    const string line = StringReplace(lines[i], "\\", "\\\\", true);
+    const string line = StripPrefixString(
+        StringReplace(lines[i], "\\", "\\\\", true), " ");
```

Comment:
decoupled the /* and */ with /\* and *\/... that makes it work in both formats... @ symbols were already covered and work for both formats... maybe not in the best way for apple docs, but otherwise xcode complains